cmd/router/main.go

```diff
@@ -98,33 +100,72 @@ func timeTrack(start time.Time, nodeOrStep string, name string) {
 	log.Info("elapsed time", nodeOrStep, name, "time", elapsed)
 }
 
-func routeStep(nodeName string, graph v1alpha1.InferenceGraphSpec, input []byte, headers http.Header) ([]byte, error) {
+type EnsembleStepOutput struct {
+	StepResponse   map[string]interface{}
+	StepStatusCode int
+}
+
+func routeStep(nodeName string, graph v1alpha1.InferenceGraphSpec, input []byte, headers http.Header) ([]byte, int, error) {
 	defer timeTrack(time.Now(), "node", nodeName)
 	currentNode := graph.Nodes[nodeName]
 
 	if currentNode.RouterType == v1alpha1.Splitter {
 		return executeStep(pickupRoute(currentNode.Steps), graph, input, headers)
 	}
 	if currentNode.RouterType == v1alpha1.Switch {
+		var statusCode int
+		var responseBytes []byte
+		var err error
 		route := pickupRouteByCondition(input, currentNode.Steps)
 		if route == nil {
-			return input, nil //TODO maybe should fail in this case?
+			errorMessage := "None of the routes matched with the switch condition"
+			log.Error(err, errorMessage)
+			/*
+				TODO Should it be 500 or 404 when none of the routes match.
+				For backward compatibility 500 is needed but 404 also looks fine since the router got bad input and didn't know what to do.
+				See what reviewers say.
+			*/
+			return nil, 500, errors.New(errorMessage)
 		}
-		return executeStep(route, graph, input, headers)
+		stepType := "serviceUrl"
```

Comment:
What's the significance of stepType? is it mostly for logging and debugging purpose?

Reply:
Yes